### PR TITLE
feat(swap): adapting the swap with an interface ISwap

### PR DESCRIPTION
**Pull request by tomrpl**
adapting the swap on the callback contract with an ISwap interface

### CODE REVIEW

File: src/blue/CallbacksSnippets.sol [hunk 1]
```diff
@@ -30,17 +31,24 @@ One should be aware that has to be taken into account on potential swap:
...
+
 contract CallbacksSnippets is IMorphoSupplyCollateralCallback, IMorphoRepayCallback, IMorphoLiquidateCallback {
     using MathLib for uint256;
     using MorphoLib for IMorpho;
     using MarketParamsLib for MarketParams;
     using SafeTransferLib for ERC20;
 
     IMorpho public immutable morpho;
-    SwapMock swapMock;
+    ISwap public swapMock;
```
Comment (julien-devatom) [inline src/blue/CallbacksSnippets.sol:42]:
```suggestion
    ISwap public immutable swapper;
```

File: src/blue/CallbacksSnippets.sol [hunk 2]
```diff
@@ -30,17 +31,24 @@ One should be aware that has to be taken into account on potential swap:
...
 
-    constructor(address morphoAddress) {
+    constructor(address morphoAddress, address swapAddress) {
         morpho = IMorpho(morphoAddress);
+        swapMock = ISwap(swapAddress);
+    }
```
Comment (julien-devatom) [inline src/blue/CallbacksSnippets.sol:47]:
```suggestion
    constructor(IMorpho _morpho, ISwap _swapper) {
        morpho = _morpho;
        swapper = _swapper;
    }
```

**Review by julien-devatom — CHANGES_REQUESTED** [review]
"mock" is an implementation, not an interface. The naming of the interface should correspond to the expected behavior of the contract. Here, it is a swapper.

**Comment by tomrpl** [comment]
Adapting tests: they are passing ✅

**Review by julien-devatom — APPROVED** [review]

**Pull request merged**